test: add basic test for journaled dao + nodeFileDao

drish opened the pull request.

diaswrd committed:
Add HttpServer nano

diaswrd committed:
Add HttpServer class

mdittmer committed:
Many-to-many DAO: Code clean up and implement find()

This creates a copy of the 7-line FilteredDAO.select() implementation so that it can be referred to directly by find(). See comments in find() for rationale.

TanayParikh committed:
check in of enum support

diaswrd committed:
Merge branch 'HttpServer'

diaswrd committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

diaswrd committed:
Create Boot and NSpec models

diaswrd committed:
Add NSpec to genjava file

diaswrd committed:
Make Boot context a ProxyX object

mdittmer committed:
ManyToManyRelationshipDAO.find(): Simplify logic and de-duplicate FilteredDAO.select()

TanayParikh committed:
Updated refinement to permit definitions in instances

Commit:
test: add basic test for journaled dao + nodeFileDao

Commit:
test: add basic test for journaled dao + nodeFileDao

kgrgreer commented on test/src/lib/node/JournaledDAO.js:
```diff
@@ -0,0 +1,75 @@
+
+    cars = foam.dao.ArrayDAO.create();
+    for ( var i = 0; i < 50; ++i ) {
+      cars.put(
```
Put to the jDAO instead, otherwise you put()'s wont be journalled.

kgrgreer commented on test/src/lib/node/JournaledDAO.js:
```diff
@@ -0,0 +1,75 @@
+         return done(err);
+       }
+       var lines = data.split('\n').filter(Boolean);
+       expect(lines.length).toBe(50);
```
You're testing an assumed implementation detail, not the contract of the DAO interface. It could pretty print the content to be more than 50 lines for write in some binary format that puts all data on one line. Create a second JournaledDAO and see if it has the data you put in the first one. You care that it's a valid DAO, not how its implemented.

kgrgreer committed:
Added U2 readme.

Commit:
test: improve journaled dao test

Commit:
test: improve journaled dao test

Commit:
test: formatting

Commit:
test: formatting

kgrgreer commented on test/src/lib/node/JournaledDAO.js:
```diff
@@ -71,5 +76,32 @@ describe('JournaledDAO', function() {
+    })
+    .select({
+      put: function(sub, data) {
+        cars.put(data);
```
This should be redundant. The JournaledDAO will already copy all of the journal data into the delegate (cars) when it's created.

kgrgreer commented on test/src/lib/node/JournaledDAO.js:
```diff
@@ -58,7 +62,8 @@ describe('JournaledDAO', function() {
   it('should persist to file', function(done) {
 
-    cars.select({ put: function(sub, car) { jDAO.put(car)}, eof: function () {} })
+    inboundDAO
```
I disagree that this whole block should exist.

kgrgreer committed:
Added README.

diaswrd committed:
Add Boot nano

TanayParikh committed:
Refactored value property to ordinal

- Added forLabel to get enum by label

TanayParikh committed:
Removed debugging code

TanayParikh committed:
Removed redundant EnumModel type definition

TanayParikh committed:
Merge branch 'master' into enum_support

kgrgreer committed:
Shorted License.

TanayParikh committed:
Removed index property

kgrgreer committed:
Changed packages.

kgrgreer committed:
Renamed EmptyContext usage to EmptyX.

TanayParikh committed:
Removed androidResource check

diaswrd committed:
Minor fix.

diaswrd committed:
Fixes on HttpServer

diaswrd committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

diaswrd committed:
Rename HttpServer

diaswrd committed:
Split http server and http handler

diaswrd committed:
Fixes on nano http server and handler

diaswrd committed:
Updates on MapDAO and Boot class

jacksonic committed:
Fixing DAOByExample putFn() parameter ordering

jacksonic committed:
Fixed DAOByExample DAO/sink adaptation

TanayParikh committed:
Updated Java Enum support through extension of foam.CLASS

- Removed template for java enum
- Replaced with FOAM2 convention for java class generation

TanayParikh committed:
Removed debugging code

kgrgreer committed:
TestRunner psedo code.

TanayParikh committed:
Refactored Java `EnumModel` to `Enum`

diaswrd committed:
Fix compilation errors on Boot class

TanayParikh committed:
Removed redundant properties

TanayParikh committed:
Removed toJavaSource function (inherited)

diaswrd committed:
Add missing primaryKey attribute on MapDAO

diaswrd committed:
Fix currentThread call error

kgrgreer committed:
Added copyright headers.

m-cheung committed:
Fix for broken console functions in examples

m-cheung committed:
Simplification to previous commit

TanayParikh committed:
Updated java/Method.js to facilitate constructor

- Constructor doesn’t have a set type
- Is there an existing best convention for this?

diaswrd committed:
Fix NSpec model definition

TanayParikh committed:
Added ability to create enum java code to foam.java.class

TanayParikh committed:
Updated foam.java.enum to utilize foam.java.class

- Increase code re-use
- Added isEnum property to distinguish enums from classes

kgrgreer committed:
Test now extends Script.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

diaswrd committed:
Fixes on NSpec model definition

kgrgreer committed:
Fix syntax error.

kgrgreer committed:
Fix syntax errors.

mdittmer committed:
foam.net.node.Handler: Helper methods to report remote IP on warn() and error()

diaswrd committed:
Fixes in Boot compilation

diaswrd committed:
Fixes on foam nanos

diaswrd committed:
Fixes on NSpec model and Boot service

diaswrd committed:
Add missing license headers.

marcogbarcellos committed:
minor tweaks on AATree Node methods

kgrgreer committed:
Merge pull request #329 from marcogbarcellos/aaTree-fix

minor tweaks on AATree Node methods

kgrgreer committed:
Many fixes.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

diaswrd committed:
Delete NSpec.java generated file

diaswrd committed:
Readd origintrails files in java_src/com/chrome

kgrgreer committed:
Merge pull request #328 from diaswrd/master

NanoServices + Boot + HttpServer

kgrgreer committed:
Fixed language reference property.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

adamvy committed:
Merge pull request #322 from TanayParikh/enum_support

Enum Support - Java Code Generation

jlhughes committed:
Add missing DAOSink requires, update COUNT API change: count -> value

Also, comment out self.pub('on', 'reset') in select as it's triggerring repeated onDAOUpdate in the caller. It's possibly a publish event per cache put, in production the event cascade never ends.

kgrgreer committed:
Merge pull request #330 from jlhughes/LazyCacheDAO

Add missing DAOSink requires, update COUNT API change: count -> value

pavneetskapoor committed:
Made changes to genjava file for external classes file and output path

kgrgreer committed:
Merge pull request #333 from pavneetskapoor/master

Made changes to genjava file for external classes file and output path

Commit:
test: improve test flow

Commit:
fix minor conflict
